pysph/sph/scheme.py

```diff
@@ -1087,7 +1088,14 @@ def get_equations(self):
         g3 = []
         for solid in self.solids:
             g3.append(WallBoundary(solid, sources=self.fluids))
-            equations.append(Group(equations=g3))
+        equations.append(Group(equations=g3))
+
+        gh = []
+        for fluid in self.fluids:
```

Comment:
The ghost stuff should be added only if a new kwarg called `has_ghosts` is set to True.  What if someone is not using ghosts?

Reply:
if someone is using the predefined scheme and has not overloaded 
    
    def setup_properties(self, particles, clean=True):
        ...

it will setup the `orig_idx`, but we can have a check for ghosts, will save some computation time checking if a particle in domain is a ghost